kubechain/internal/controller/taskrun/taskrun_controller.go

```diff
@@ -420,24 +421,25 @@ func (r *TaskRunReconciler) endTaskRunSpan(ctx context.Context, taskRun *kubecha
 func (r *TaskRunReconciler) processLLMResponse(ctx context.Context, output *kubechainv1alpha1.Message, taskRun *kubechainv1alpha1.TaskRun, statusUpdate *kubechainv1alpha1.TaskRun) (ctrl.Result, error) {
 	logger := log.FromContext(ctx)
 
-	if output.Content != "" {
-		// final answer branch
-		statusUpdate.Status.Output = output.Content
-		statusUpdate.Status.Phase = kubechainv1alpha1.TaskRunPhaseFinalAnswer
-		statusUpdate.Status.Ready = true
-		statusUpdate.Status.ContextWindow = append(statusUpdate.Status.ContextWindow, kubechainv1alpha1.Message{
-			Role:    "assistant",
-			Content: output.Content,
-		})
-		statusUpdate.Status.Status = StatusReady
-		statusUpdate.Status.StatusDetail = "LLM final response received"
-		statusUpdate.Status.Error = ""
-		r.recorder.Event(taskRun, corev1.EventTypeNormal, "LLMFinalAnswer", "LLM response received successfully")
+	// Log complete output message for debugging
+	outputBytes, _ := json.Marshal(output)
+	logger.Info("DEBUG: Processing LLM response", "outputMessage", string(outputBytes))
 
-		// End the parent span since we've reached a terminal state
-		r.endTaskRunSpan(ctx, taskRun, codes.Ok, "TaskRun completed successfully with final answer")
-	} else {
+	// Check if we have any tool calls
+	hasToolCalls := len(output.ToolCalls) > 0
+	logger.Info("DEBUG: LLM response stats", "hasContent", output.Content != "", "hasToolCalls", hasToolCalls, "toolCallCount", len(output.ToolCalls))
+
+	if hasToolCalls {
 		// tool call branch: create TaskRunToolCall objects for each tool call returned by the LLM.
+		logger.Info("DEBUG: Taking tool calls branch because tool calls are present", "contentPresent", output.Content != "")
```

Comment:
does this work?
```suggestion
		logger.Debug("Taking tool calls branch because tool calls are present", "contentPresent", output.Content != "")
```

Reply:
(not a blocker)